modules/messages/src/lib.rs

```diff
@@ -807,16 +807,14 @@ impl<T: Config<I>, I: 'static> RuntimeInboundLaneStorage<T, I> {
 	/// `receive_messages_proof` call, because the actual inbound lane state is smaller than the
 	/// maximal configured.
 	///
-	/// Maximal inbound lane state set size is configured by the
-	/// `MaxUnrewardedRelayerEntriesAtInboundLane` constant from the pallet configuration. The PoV
-	/// of the call includes the maximal size of inbound lane state. If the actual size is smaller,
-	/// we may subtract extra bytes from this component.
+	/// Maximal inbound lane state size is computed using the
+	/// `MaxUnrewardedRelayerEntriesAtInboundLane` and `MaxUnconfirmedMessagesAtInboundLane`
+	/// constants from the pallet configuration. The PoV of the call includes the maximal size
+	/// of the inbound lane state. If the actual size is smaller, we may subtract extra bytes
+	/// from this component.
 	pub fn extra_proof_size_bytes(&self) -> u64 {
 		let max_encoded_len = StoredInboundLaneData::<T, I>::max_encoded_len();
-		let relayers_count = self.data().relayers.len();
-		let actual_encoded_len =
-			InboundLaneData::<T::InboundRelayer>::encoded_size_hint(relayers_count)
-				.unwrap_or(usize::MAX);
+		let actual_encoded_len = self.data().encoded_size();
```

Comment:
I'd still use `encoded_size_hint`. `encoded_size` does real encoding, which takes time and do allocations. And we're in runtime here => performance really matters

Reply:
Done